pkg/monitor/cluster/cluster.go

```diff
@@ -92,7 +93,15 @@ func NewMonitor(log *logrus.Entry, restConfig *rest.Config, oc *api.OpenShiftClu
 		return nil, err
 	}
 
-	ocpclientset, err := client.New(restConfig, client.Options{})
+	// lazy discovery will not attempt to reach out to the apiserver immediately
+	mapper, err := apiutil.NewDynamicRESTMapper(restConfig, apiutil.WithLazyDiscovery)
```

Comment:
I am not sure this will deal with the underlying issue of death kube-apiserver. The DynamicRESTMapper first talks to the API to get the resources during creation time. The calls that follow are:

- https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go#L85-L86 
- https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/k8s.io/client-go/restmapper/discovery.go#L148
- https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/k8s.io/client-go/discovery/discovery_client.go#L216
- https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/k8s.io/client-go/discovery/discovery_client.go#L244
- https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/k8s.io/client-go/discovery/discovery_client.go#L161

The laze loading kicks in later on.

Reply:
@petrkotas in our monitoring stack, we run a series of checks.  [The first check is to call the apiserver](https://github.com/Azure/ARO-RP/blob/master/pkg/monitor/cluster/cluster.go#L147-L161).  If that call fails, the rest of the checks will not run as we return early and emit a metric.  

We don't actually care if the apiserver is up during client construction, what we do care about is that we're able to successfully construct the clients in order to check if the apiserver is up and emit the metric as the first portion of our monitor.  

If the apiserver is down, all other checks will fail anyway and client construction isn't relevant.

Reply:
>The DynamicRESTMapper first talks to the API to get the resources during creation time.
If you look at that piece of code linked, it's actually not calling the apiserver.  [It's creating a function which would call the apiserver to discover resources](https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go#L82-L91).  The actual call to the apiserver happens within the [`setStaticMapper` function](https://github.com/bennerv/ARO-RP/blob/375cc54a2964d979ecf0b4e831bcc64b730fbbfb/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/dynamicrestmapper.go#L98).